src/bionics.cpp

```diff
@@ -1336,6 +1337,14 @@ void player::bionics_uninstall_failure( int difficulty, int success, float adjus
         case 3:
             for( const body_part &bp : all_body_parts ) {
                 if( has_effect( effect_under_op, bp ) ) {
+                    if( is_parent_bp_in_set( bp, bp_hurt ) ) {
```

Comment:
The data you stashed in is_parent_bp_in_set() already exists in the body part data, can use it like so:
```suggestion
                    if( bp_hurt.count( mutate_to_main_part( bp ) > 0 ) ) ) {
```